server/src/main/java/org/opensearch/action/search/TransportDeletePitAction.java

```diff
@@ -0,0 +1,98 @@
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Transport action for deleting point in time searches - supports deleting list and all point in time searches
+ */
+public class TransportDeletePitAction extends HandledTransportAction<DeletePitRequest, DeletePitResponse> {
+    private final NamedWriteableRegistry namedWriteableRegistry;
+    private TransportSearchAction transportSearchAction;
+    private final ClusterService clusterService;
+    private final SearchTransportService searchTransportService;
+
+    @Inject
+    public TransportDeletePitAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedWriteableRegistry namedWriteableRegistry,
+        TransportSearchAction transportSearchAction,
+        ClusterService clusterService,
+        SearchTransportService searchTransportService
+    ) {
+        super(DeletePitAction.NAME, transportService, actionFilters, DeletePitRequest::new);
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportSearchAction = transportSearchAction;
+        this.clusterService = clusterService;
+        this.searchTransportService = searchTransportService;
+    }
+
+    /**
+     * Invoke 'delete all pits' or 'delete list of pits' workflow based on request
+     */
+    @Override
+    protected void doExecute(Task task, DeletePitRequest request, ActionListener<DeletePitResponse> listener) {
+        List<String> pitIds = request.getPitIds();
+        if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
+            deleteAllPits(listener);
+        } else {
+            deletePits(listener, request);
+        }
+    }
+
+    /**
+     * Deletes one or more point in time search contexts.
+     */
+    private void deletePits(ActionListener<DeletePitResponse> listener, DeletePitRequest request) {
+        Map<String, List<PitSearchContextIdForNode>> nodeToContextsMap = new HashMap<>();
+        for (String pitId : request.getPitIds()) {
+            SearchContextId contextId = SearchContextId.decode(namedWriteableRegistry, pitId);
+            for (SearchContextIdForNode contextIdForNode : contextId.shards().values()) {
+                PitSearchContextIdForNode pitSearchContext = new PitSearchContextIdForNode(pitId, contextIdForNode);
+                List<PitSearchContextIdForNode> contexts = nodeToContextsMap.getOrDefault(contextIdForNode.getNode(), new ArrayList<>());
+                contexts.add(pitSearchContext);
+                nodeToContextsMap.put(contextIdForNode.getNode(), contexts);
+            }
+        }
+        SearchUtils.deletePitContexts(nodeToContextsMap, listener, clusterService.state(), searchTransportService);
+    }
+
```

Comment:
Lets move this to a Service

Reply:
Made the change to make it a separate service. One challenge with just putting everything in create pit controller is that there are too many dependencies and integ tests fail with inject errors ( sometimes even on the classes which are existing ). So created a new lightweight service file which will have the common methods. Create pit controller will continue to cater create pit specific methods.